datadog/dogshell/monitor.py

```diff
@@ -173,6 +207,100 @@ def _show_all(cls, args):
                                  (d["query"]),
                                  (d["type"])]))
 
+    @classmethod
+    def _pull(cls, args):
+        api._timeout = args.timeout
+        res = api.Monitor.get(args.monitor_id)
+        report_warnings(res)
+        report_errors(res)
+
+        cls._write_monitor_to_file(
+            res, args.filename,
+            args.format, args.string_ids)
+
+    @classmethod
+    def _pull_all(cls, args):
+        api._timeout = args.timeout
+
+        def _name_to_filename(name):
+            # Get a lowercased version with most punctuation stripped out...
+            no_punct = ''.join([c for c in name.lower() if c.isalnum() or c in [" ", "_", "-"]])
+            # Now replace all -'s, _'s and spaces with "_", and strip trailing _
+            return no_punct.replace(" ", "_").replace("-", "_").strip("_")
+
+        format = args.format
+        res = api.Monitor.get_all(
```

Comment:
This won’t work for orgs with lots of monitors as it will auto paginate. At super scale there is also an alternative pagination method, I’ve implemented in a different tool in this PR: https://github.com/airbnb/interferon/pull/58